src/main/java/fi/metatavu/kuntaapi/server/integrations/ptv/translation/KuntaApiPtvTranslator.java

```diff
@@ -463,36 +463,40 @@
   /**
    * Translates Kunta API delivery address into PTV in addresses
    * 
+   * @param formReceiver 
    * @param deliveryAddress Kunta API delivery address
    * @return PTV in addresses
    */
-  public V7VmOpenApiAddressDeliveryIn translateDeliveryAddresses(Address deliveryAddress) {
-    if (deliveryAddress == null) {
-      return null;
+  public List<V8VmOpenApiAddressDeliveryIn> translateDeliveryAddresses(List<LocalizedValue> formReceiver, Address deliveryAddress) {
+    if (deliveryAddress == null && (formReceiver == null || formReceiver.isEmpty())) {
+      return Collections.emptyList();
     }
     
-    V7VmOpenApiAddressDeliveryIn result = new V7VmOpenApiAddressDeliveryIn(); 
-    
-    PtvAddressSubtype subtype = getAddressSubtype(deliveryAddress.getSubtype());
-    switch (subtype) {
-      case POST_OFFICE_BOX:
-        result.setPostOfficeBoxAddress(translatePostOfficeBoxAddress(deliveryAddress));
-      break;
-      case NO_ADDRESS:
-        result.setDeliveryAddressInText(translateLocalizedValuesIntoLanguageItems(deliveryAddress.getAdditionalInformations()));
-      break;
-      case SINGLE:
-      case STREET:
-        result.setStreetAddress(translateStreetAddress(deliveryAddress));
-      break;
-      default:
-        logger.log(Level.SEVERE, () -> String.format(UNKNOWN_SUBTYPE, deliveryAddress.getSubtype()));
-      break;
+    V8VmOpenApiAddressDeliveryIn result = new V8VmOpenApiAddressDeliveryIn(); 
+    
+    if (deliveryAddress != null) {
+      PtvAddressSubtype subtype = getAddressSubtype(deliveryAddress.getSubtype());
+      switch (subtype) {
+        case POST_OFFICE_BOX:
+          result.setPostOfficeBoxAddress(translatePostOfficeBoxAddress(deliveryAddress));
+        break;
+        case NO_ADDRESS:
+          result.setDeliveryAddressInText(translateLocalizedValuesIntoLanguageItems(deliveryAddress.getAdditionalInformations()));
+        break;
+        case SINGLE:
+        case STREET:
+          result.setStreetAddress(translateStreetAddress(deliveryAddress));
+        break;
+        default:
+          logger.log(Level.SEVERE, () -> String.format(UNKNOWN_SUBTYPE, deliveryAddress.getSubtype()));
+        break;
+      }
     }
 
     result.setSubType(deliveryAddress.getSubtype());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "deliveryAddress" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)